### PR TITLE
Tha cuber treeview copy file location

### PR DESCRIPTION
**Pull request by SwissalpS**
if we don't check settings, we may be force-loading a plugin that user
did not want active.

### CODE REVIEW

File: plugins/copyfilelocation.lua [hunk 1]
```diff
@@ -1,9 +1,15 @@
 -- mod-version:3
 local core = require "core"
 local command = require "core.command"
-local treeview = require 'plugins.treeview'
-local treemenu = treeview.contextmenu
-local mainmenu = require 'plugins.contextmenu'
+local config = require "core.config"
+local treeview, treemenu, mainmenu
+if false ~= config.plugins.treeview then
```
Comment (thacuber2a03) [inline plugins/copyfilelocation.lua:6]: this hurts me /lh

File: plugins/copyfilelocation.lua [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
+  treeview = require "plugins.treeview"
+end
+if false ~= config.plugins.contextmenu then
```
Comment (thacuber2a03) [inline plugins/copyfilelocation.lua:9]: this also does (still /lh)

**Review by thacuber2a03 — COMMENTED** [review]
the `false ~=` lines aren't really "Lua" styled, but that's just me
everything else looks okay

**Comment by SwissalpS** [comment]
> the `false ~=` lines aren't really "Lua" styled, but that's just me everything else looks okay

it could be ``nil`` when user has not set a preference, that is why we need to check against ``false`` explicitly

File: plugins/copyfilelocation.lua (hunk 1 above)
Comment (SwissalpS) [reply plugins/copyfilelocation.lua:6]: I don't understand "/lh" so I don't get what is hurting you.

Anyway, just a suggestion, if you see a better way to do it, go ahead :D

**Comment by thacuber2a03** [comment]
well but `false` isn't usually put before the `~=`

**Comment by SwissalpS** [comment]
most code also uses " and not ' as you and I both do :D (I changed that)

these things have nothing to do with lua, they are personal preferences (in lua) although there are reasons to put ``false`` first. I won't go into those reasons here, there are sufficent books out there explaining the reasons for that.

File: plugins/copyfilelocation.lua (hunk 1 above)
Comment (thacuber2a03) [reply plugins/copyfilelocation.lua:6]: oh, heh, it means lighthearted, so it doesn't hurt me in the personal sense, just a small annoyance

**Comment by SwissalpS** [comment]
completely forgot about this. There is actually a slightly better way to respect user settings, I should have updated that on this branch.

Hopefully I don't forget again and will post later today.

**Comment by thacuber2a03** [comment]
> completely forgot about this.

can't lie, I also did

**Comment by SwissalpS** [comment]
there we go: #2

**Pull request merged**